decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -133,6 +124,18 @@ def maximum_votes_reached?
         votes.count >= maximum_votes
       end
 
+      # Public: Instantiates a presenter for the author of the proposal and yields
+      #   the given block passing the instantiated presenter to it.
+      #
+      # Returns the instantiated presenter
+      def present_author
```

Comment:
What do you think about moving this logic to a helper method? You could refactor the `present` method in `ApplicationHelper` in `core` to accept a specific presenter class as an optional parameter, , then in proposals define a helper method called `present_proposal_author` that calculates the logic to get the presenter class name and use the `presenter`method.

This way we would keep the same API in both `core` and `proposals`, and stop coupling models and views 😄

Reply:
Yeah! Also, coupling a model to view logic doesn't sound good to me. 

You could instead inject the view context in the presenter via the helper, so we don't need to do weird tricks by including routes & everything manually - anything view-related could be rendered by calling `view.resource_url` or similar.

Reply:
Ooops, I wanted to do this but I forgot, the idea was to do `present(proposal) do` in the view and instantiate the presented author from that presenter class. Does that sound good?

Reply:
Ah yes, that sounds really good indeed!